src/traces/mesh3d/convert.js

```diff
@@ -52,7 +52,53 @@ proto.handlePick = function(selection) {
 };
 
 function parseColorArray(colors) {
-    return colors.map(str2RgbaArray);
+    var b = [];
+    var len = colors.length;
+    for(var i = 0; i < len; i++) {
+        b[i] = str2RgbaArray(colors[i]);
+    }
+    return b;
+}
+
+// Unpack position data
```

Comment:
@etpinard wondering where could these functions go if we want to reuse these functions in other places e.g. in iso-surfaces?

Reply:
I'd vote for putting helper functions like this one in a new file `src/traces/mesh3d/helpers.js`.